### PR TITLE
Start of revamp

### PR DESCRIPTION
**Pull request by slacAWallace**
Changed FFO to utilize internal array for all fast faults. Fast fault FB now just grabs a spot in that array at init, fills in meta info, and updates with state of fast fault logic.

FFO takes care of veto and reset logic.

All FF controls and readbacks are accessible from FFO array.

### CODE REVIEW

**Comment by slacAWallace** [comment]
If it looks copacetic then I'll be merging.

File: lcls-twincat-pmps/lcls-twincat-pmps/Library/DUTs/ST_FF.TcDUT [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    OK   :   BOOL; // Fault logic state
+    
+    {attribute 'pytmc' := '
+        pv: BeamPermitted
+        io: i
+     '}
+    BeamPermitted   :   BOOL; // Result of reset, veto, and fault logic, true beam off boolean
+    
+    {attribute 'pytmc' := '
+        pv: OK
```
Comment (ZLLentz) [inline lcls-twincat-pmps/lcls-twincat-pmps/Library/DUTs/ST_FF.TcDUT]: copy/paste error?

File: lcls-twincat-pmps/lcls-twincat-pmps/Library/DUTs/ST_FFVeto.TcDUT [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <Declaration><![CDATA[TYPE ST_FFVeto :
+STRUCT
+    {attribute 'pytmc' := '
+        pv: Duration
+        io: o
+     '}
+    Duration : TIME;
+    
+    {attribute 'pytmc' := '
+        pv: Duration
```
Comment (ZLLentz) [inline lcls-twincat-pmps/lcls-twincat-pmps/Library/DUTs/ST_FFVeto.TcDUT]: copy/paste error 2?

**Comment by ZLLentz** [comment]
It's hard for me to parse the functionality of this but I pointed out the things that seemed wrong

**Comment by slacAWallace** [comment]
@ZLLentz Thanks for catching those errors. Should be fixed now.

**Pull request merged**